lib/src/css_box_widget.dart

```diff
@@ -0,0 +1,712 @@
+        firstChild, (RenderBox child) => child.getMaxIntrinsicWidth(height));
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    return getIntrinsicDimension(
+        firstChild, (RenderBox child) => child.getMinIntrinsicHeight(width));
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    return getIntrinsicDimension(
+        firstChild, (RenderBox child) => child.getMaxIntrinsicHeight(width));
+  }
+
+  @override
+  double? computeDistanceToActualBaseline(TextBaseline baseline) {
+    return firstChild?.getDistanceToActualBaseline(baseline);
+  }
+
+  @override
+  Size computeDryLayout(BoxConstraints constraints) {
+    return _computeSize(
+      constraints: constraints,
+      layoutChild: ChildLayoutHelper.dryLayoutChild,
+    ).parentSize;
+  }
+
+  _Sizes _computeSize(
+      {required BoxConstraints constraints,
+      required ChildLayouter layoutChild}) {
+    if (childCount == 0) {
+      return _Sizes(constraints.biggest, Size.zero);
+    }
+
+    Size containingBlockSize = constraints.biggest;
+    double width = containingBlockSize.width;
+    double height = containingBlockSize.height;
+
+    RenderBox? child = firstChild;
+    assert(child != null);
+
+    // Calculate child size
+    final childConstraints = constraints.copyWith(
+      maxWidth: (this.width.unit != Unit.auto)
+          ? this.width.value
+          : containingBlockSize.width -
+              (this.margins.left?.value ?? 0) -
+              (this.margins.right?.value ?? 0),
+      maxHeight: (this.height.unit != Unit.auto)
+          ? this.height.value
+          : containingBlockSize.height -
+              (this.margins.top?.value ?? 0) -
+              (this.margins.bottom?.value ?? 0),
+      minWidth: (this.width.unit != Unit.auto) ? this.width.value : 0,
+      minHeight: (this.height.unit != Unit.auto) ? this.height.value : 0,
+    );
+    final Size childSize = layoutChild(child!, childConstraints);
+
+    // Calculate used values of margins based on rules
+    final usedMargins = _calculateUsedMargins(childSize, containingBlockSize);
+    final horizontalMargins =
+        (usedMargins.left?.value ?? 0) + (usedMargins.right?.value ?? 0);
+    final verticalMargins =
+        (usedMargins.top?.value ?? 0) + (usedMargins.bottom?.value ?? 0);
+
+    //Calculate Width and Height of CSS Box
+    height = childSize.height;
+    switch (display) {
+      case Display.BLOCK:
+        width = (shrinkWrap || childIsReplaced)
+            ? childSize.width + horizontalMargins
+            : containingBlockSize.width;
+        height = childSize.height + verticalMargins;
+        break;
+      case Display.INLINE:
+        width = childSize.width + horizontalMargins;
+        height = childSize.height;
+        break;
+      case Display.INLINE_BLOCK:
+        width = childSize.width + horizontalMargins;
+        height = childSize.height + verticalMargins;
+        break;
+      case Display.LIST_ITEM:
```

Comment:
Just a thought: This might not be sufficient for list items that (via css) you force to a specific `display: inline`?

Reply:
I'm going to push this one off as future work. We've got a whole slew of list issues that need to be resolved so I'll tackle this when I start tackling those.